src/ai/backend/manager/models/base.py

```diff
@@ -191,6 +194,32 @@ def copy(self):
         return ResourceSlotColumn()
 
 
+class StructuredJSONBColumn(TypeDecorator):
+    """
+    A column type check scheduler_opts's validation using trafaret.
+    """
+
+    impl = JSONB
+    cache_ok = True
+
+    def process_bind_param(self, value, dialect):
+        return value if value is not None else None
+
+    def process_result_value(self, raw_value, dialect):
+        # legacy handling
+        schedulerOption = t.Dict(
+            {
+                tx.AliasedKey(['allowed_session_type', 'session_type', 'sessionType'],
+                                default='interactive') >> 'allowed_session_type': tx.Enum(SessionTypes),
+            }).allow_extra('*')
```

Comment:
This should be provided at the time of instantiation (i.e., the model field definitions), not hard-coded inside the class.

Reply:
At first, I received the trafaret object as a parameter of StructuredJSONBColumn, but when I tested it, an error occurred
![dialector](https://user-images.githubusercontent.com/18081105/150511167-2a80aafa-14b7-4c5a-808a-b011c6198bf9.png)
![dialector2](https://user-images.githubusercontent.com/18081105/150511178-4b8ecd79-6cd4-4421-875d-043c7fd765bc.png)
.

Reply:
If you pass a trafaret object when **instantiating** the column class, it should work as expected.

Example:
```python
    sa.Column('scheduler_opts', StructuredJSONBColumn(
        t.Dict({
            t.Key('allowed_session_types', default=['interactive', 'batch']):
                t.List(tx.Enum(SessionTypes)),
        }).allow_extra('*'),
    ), nullable=False, default={}),
```

Reply:
Also there is no need to use `tx.AliasedKey` because it is for legacy clients which use different names for parameter keys.

Reply:
When I edit source code, I referred to task_template_v1 in the session_template.py, but I didn't fix it according to the allowed_session_types...😥